Dockerfile.gpu

```diff
@@ -82,8 +82,9 @@ RUN if [ "$TORCHVISION_PIP" ]; then ${CONDA_DIR}/bin/python3.6 -m pip install $T
 RUN if [ "$TENSORPACK_PIP" ]; then ${CONDA_DIR}/bin/python3.6 -m pip install $TENSORPACK_PIP; fi
 RUN ${CONDA_DIR}/bin/python3.6 -m pip install GPUtil
 
-ARG TORCH_CUDA_ARCH_LIST="3.7;6.0;6.1;6.2;7.0;7.5"
-RUN if [ "$TORCH_PIP" ]; then pip install --global-option="--cpp_ext" --global-option="--cuda_ext" git+https://github.com/determined-ai/apex.git@37cdaf4ad57ab4e7dd9ef13dbed7b29aa939d061; fi
+ARG TORCH_CUDA_ARCH_LIST
+ARG APEX_GIT
+RUN if [ "$APEX_GIT" ]; then pip install --global-option="--cpp_ext" --global-option="--cuda_ext" git+$APEX_GIT; fi
```

Comment:
Why parameterize this if we always set it for all gpu images?  (or do we?)

Reply:
Parameterizing this because the newest upstream version does not like being built in a CUDA 11 environment, but there are CUDA 11-specific fixes in it. So the CUDA 10 images will get our downstream (although maybe we should rebase it and only remove fixes we have cause to omit) and the CUDA 11 images will get the newer upstream.

Reply:
Well, that's why I'm parameterizing APEX_GIT. TORCH_CUDA_ARCH_LIST has a similar issue where it can't contain the new compute architecture when you're building for CUDA 10, but it MUST be set when you're building for CUDA 11. Failure to do this results in it acting like you have no GPUs.